src/stepfunctions/steps/integration_resources.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from __future__ import absolute_import
+
+from enum import Enum
+
+"""
+Enum classes for task integration resource arn builder
+"""
+
+
+class IntegrationPattern(Enum):
+    WaitForTaskToken = "waitForTaskToken"
+    WaitForCompletion = "sync"
+
+
+class IntegrationServices(Enum):
+    Lambda = "lambda"
+    SageMaker = "sagemaker"
+    Glue = "glue"
+    ECS = "ecs"
+    Batch = "batch"
+    DynamoDB = "dynamodb"
+    SNS = "sns"
+    SQS = "sqs"
+    ElasticMapReduce = "elasticmapreduce"
+
+
+class LambdaApi(Enum):
+    Invoke = "invoke"
```

Comment:
Let's remove these. We don't need to refer to the service or api names outside the Step classes.

Reply:
I had similar thought before I start coding. 
But during the implementation, I found it's important to have the well organized service/api for development. It will not only help with auto-completion which helps accelerate coding speed, but will also help reading/changing/examining the code as we have our resource gathered in a place rather than sparsely distributed among places.
Another reason is: we have the integration pattern passed in as enum, this is limited number hard-coded param exactly same as our integrated services and their apis. 
I think this will also help with organizing further expansion when we are introducing more service and apis into the repo.

Reply:
I'm not convinced. Arguments for types and autocompletion make sense if they are to be used in many places. That is not the case for "lambda" and "invoke", etc.. There's no need to reference them outside the module defining the service integration steps, so don't expose them.

> Another reason is: we have the integration pattern passed in as enum, this is limited number hard-coded param exactly same as our integrated services and their apis.

Not sure what you mean here